libraries/AP_Arming/AP_Arming.h

```diff
@@ -166,6 +173,22 @@ class AP_Arming {
         MIS_ITEM_CHECK_RALLY         = (1 << 5),
         MIS_ITEM_CHECK_MAX
     };
+
+    // external authorisation
+    static const uint8_t ext_auth_count_max = 3;
+    static const uint8_t ext_auth_str_len = 40;
+    enum class ExtAuthStates {
+        AUTH_STATE_UNASSIGNED = 0,
+        AUTH_STATE_NO_RESPONSE,
+        AUTH_STATE_FAILED,
+        AUTH_STATE_PASSED
+    };
+    struct ext_auth_t {
+        ExtAuthStates state;
+        char* fail_msg;
+    };
+    ext_auth_t ext_auth[ext_auth_count_max] = {};
```

Comment:
I don't think using a bitmask is worth the effort or additional complexity it would add.  The structure is only 7 bytes per authoriser so there's very limited room for shrinking the basic structure.
I'll switch it to a single failure message buffer but we should note that this comes at the cost of not being able to display all arming errors at once from independent scripts or companion computers.  I think it's very possible we could get an enhancement request to show all the failures at once

Reply:
6 bytes total would support 32 arming checks with my proposal though... Using the proposed structure here would be 224 bytes to support that many checks. 3 checks is just not enough for arming checks as far as I can tell, I'd easily like to have more then that many for scripts.

Reply:
No, I think if we switch it to a single failure message and make the AuxAuthStates class based on a uint8_t it will consume just 32 bytes for 32 authorisers..

Reply:
While modifying this to use a single fail_msg buffer, one issue is this situation:

1. authoriser 1 fails, set failure msg to "fail 1"
2. authoriser 2 fails, sets failure msg to "fail 2"
3. authoriser 2 passes

In this situation, authoriser 1 has failed but we've lost it's failure message but we need to tell the user that we've failed arming.. but can no longer say why